Code/Common/src/sitkTransform.cxx

```diff
@@ -128,19 +128,16 @@ bool initialized = RegisterMoreTransforms<2>() && RegisterMoreTransforms<3>();
 //
 
 Transform::Transform( )
-  : m_PimpleTransform( nullptr )
+  : m_PimpleTransform( new PimpleTransform<itk::IdentityTransform< double, 3 > >() )
```

Comment:
Please consider `std::make_unique<itk::IdentityTransform< double, 3 > >` instead. The idea is that it's best to avoid _both_ low-level memory management commands, `delete` _and_ `new`, when possible.

Reply:
The code should actually be: `std::make_unique< PimpleTransform<itk::IdentityTransform< double, 3 > > >()` which is getting a bit longer, and uglier.

> The idea is that it's best to avoid both low-level memory management commands, delete and new, when possible.

There appears to be less benefits for for this change than using the std smart pointers. Please keep improvements to genuine improvements  not just following current C++ "ideas".

Reply:
@blowekamp Seriously, manually doing `delete` _and_ `new` is a major cause of bugs in old C++ code. Please re-consider: _Use `make_unique()` to construct objects owned by `unique_ptr`s_, https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rh-make_unique

Reply:
There are many benefits of using object for handle the `delete` with object scope and exception handling.

But the linked reasoning for `make_unique` is rather poor:
https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rr-make_unique

And the resulting code style is convoluted.